ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/interceptor/AuthorizationInterceptor.scala

```diff
@@ -53,12 +51,8 @@ final class AuthorizationInterceptor(protected val authService: AuthService, ec:
             logger.warn(s"Failed to get claims from request metadata: ${exception.getMessage}")
             call.close(internalAuthenticationError, new Metadata())
             new ServerCall.Listener[Nothing]() {}
-          case Success(Claims.empty) =>
-            logger.debug(s"Auth metadata decoded into empty claims, returning UNAUTHENTICATED")
-            call.close(Status.UNAUTHENTICATED, new Metadata())
-            new ServerCall.Listener[Nothing]() {}
-          case Success(claims) =>
-            val nextCtx = prevCtx.withValue(contextKeyClaim, claims)
+          case Success(claimSet) =>
+            val nextCtx = prevCtx.withValue(AuthorizationInterceptor.contextKeyClaimSet, claimSet)
```

Comment:
We need to return a `ServerCall.Listener[Nothing]()` when the authentication failed, instead of a listener with context

Reply:
Please correct me if I'm wrong but this way we would lose all the information from the context. The idea of changes in this PR is to deliberately allow even unauthenticated calls at the interceptor level, enrich the context with such information and let service authorisation wrappers decide if the `ClaimSet` from the context of a call is sufficient to call a method.
Side note: Maybe it would be worth renaming `AuthorizationInterceptor` to `AuthenticationInterceptor`.

Reply:
You are right about that.